src/textual/_xterm_parser.py

```diff
@@ -129,25 +131,42 @@ def parse(self, on_token: TokenCallback) -> Generator[Awaitable, str, None]:
                         if len(peek_buffer) == 1 and not more_data():
                             continue
 
+                # Look ahead through the suspected escape sequence for a match
                 while True:
-                    sequence += yield read1()
+
+                    # If we run into another ESC at this point, then we've failed
+                    # to find a match, and should issue everything we've seen within
+                    # the suspected sequence as Key events instead.
+                    buffer = yield self.peek_buffer()
```

Comment:
This doesn't look right. `peek_buffer` may return empty string if there is nothing in the buffer, but it could still read an `ESC` on L155

Reply:
I'll think about how I can get a test to cover this scenario and fix it on Tuesday.